src/server.rs

```diff
@@ -312,16 +311,16 @@ async fn race2<T, A: Future<Output = T>, B: Future<Output = T>>(f1: A, f2: B) ->
     }
 }
 
-/// Spawn a server loop, invoking a handler callback for each request.
+/// Run a server loop, invoking a handler callback for each request.
 ///
 /// Requests will be handled sequentially.
-pub async fn spawn_server<S, C, T, F, Fut>(
+pub async fn run_server_loop<S, C, T, F, Fut>(
```

Comment:
While improving the split example to add some more logging, I realized that spawning the loop in the library function is a bad idea because it makes reporting on errors harder. Instead I changed it so that it just runs the loop, returning on errors. The task spawning is done at call site instead.